src/daemon/daemon.c

```diff
@@ -472,6 +475,8 @@ static void notify_daemon_init(NotifyDaemon* daemon)
 	daemon->idle_reposition_notify_ids = g_hash_table_new(NULL, NULL);
 	daemon->monitored_window_hash = g_hash_table_new(NULL, NULL);
 	daemon->notification_hash = g_hash_table_new_full(g_int_hash, g_int_equal, g_free, (GDestroyNotify) _notify_timeout_destroy);
+
+	add_exit_timeout_if_needed(daemon);
```

Comment:
Must be moved to after `daemon->notification_hash` is initialized, as the function will check its size.